app/controllers/users_controller.rb

```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  before_filter :authenticate_user!
+
+  def show
+    @user = User.find(params[:id])
```

Comment:
Please use [before_action](http://guides.rubyonrails.org/action_controller_overview.html#highlighter_842862) to set the user is all corresponding actions.